lib/mongo/protocol.ex

```diff
@@ -27,12 +27,18 @@ defmodule Mongo.Protocol do
   defp connect(opts, s) do
     # TODO: with/else in elixir 1.3
     result =
-      with {:ok, s} <- tcp_connect(opts, s),
-           {:ok, s} <- wire_version(s),
-           {:ok, s} <- Mongo.Auth.run(opts, s) do
+      with {:ok, s} <- tcp_connect(opts, s) do
+        inner_result = if opts[:skip_auth] do
```

Comment:
This is a bit hard to read with the `with`, `if` and `else`.  Try putting the if on a new line after `inner_result =` or break into a function.